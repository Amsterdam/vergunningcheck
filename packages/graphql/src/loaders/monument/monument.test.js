const uuid = require("uuid");
const monument = require("rewire")("./monument.js");
const loader = monument.__get__("loader");

const load = jest.fn(
  (id) => new Promise((resolve) => resolve(`data for ${id}`))
);

// Setup mock
monument.__set__(
  "loader",
  Object.assign({}, loader, {
    load,
  })
);

describe("Monument loader", () => {
  beforeEach(async () => {
    jest.clearAllMocks();
  });
  test("load function", async () => {
    const key = uuid();
    const result = await Promise.all(await monument.load([key]));
    expect(result).toStrictEqual([`data for ${key}`]);
  });

  test("cache hit", async () => {
    const key = uuid();
    const a = await Promise.all(await monument.load([key]));
    const b = await Promise.all(await monument.load([key]));
    expect(a).toStrictEqual(b);
    expect(load.mock.calls.length).toBe(1);
  });

  test("cache miss", async () => {
    const a = await Promise.all(await monument.load([uuid()]));
    const b = await Promise.all(await monument.load([uuid()]));
    expect(load.mock.calls.length).toBe(2);
    expect(a).not.toStrictEqual(b);
  });
});
