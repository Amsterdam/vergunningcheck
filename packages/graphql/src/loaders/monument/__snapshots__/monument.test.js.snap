// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monument loader Basic functions module result 1`] = `
Array [
  Object {
    "_type": "Monument",
    "name": "Rijksmonument",
    "scope": "NATIONAL",
    "type": "NATIONAL_MONUMENT",
  },
]
`;

exports[`Monument loader Basic functions module result 2`] = `
Array [
  Object {
    "_type": "Monument",
    "name": "Gemeentelijk monument",
    "scope": "MUNICIPAL",
    "type": "MUNICIPAL_MONUMENT",
  },
]
`;
